test/framework/src/main/java/org/elasticsearch/test/rest/yaml/section/ClientYamlTestSuite.java

```diff
@@ -132,27 +135,52 @@ public SetupSection getSetupSection() {
         return setupSection;
     }
 
-    public void setSetupSection(SetupSection setupSection) {
-        this.setupSection = setupSection;
-    }
-
     public TeardownSection getTeardownSection() {
         return teardownSection;
     }
 
-    public void setTeardownSection(TeardownSection teardownSection) {
-        this.teardownSection = teardownSection;
+    public void validate() {
+        validateExecutableSections(this, setupSection.getExecutableSections(), null, setupSection, null);
+        validateExecutableSections(this, teardownSection.getDoSections(), null, null, teardownSection);
+        for (ClientYamlTestSection testSection : testSections) {
+            validateExecutableSections(this, testSection.getExecutableSections(), testSection, setupSection, teardownSection);
+        }
+    }
+
+    private static void validateExecutableSections(ClientYamlTestSuite yamlTestSuite, List<ExecutableSection> sections,
+                                                   ClientYamlTestSection testSection,
+                                                   SetupSection setupSection, TeardownSection teardownSection) {
+        for (ExecutableSection section : sections) {
```

Comment:
suggestion: This might read better with `.stream().filter().map()`

Reply:
you mean to have the instanceof check in the filter? I am not sure I follow 100% what you are suggesting besides that.

Reply:
Yes, the instance-of in the filter and the cast in the map at a minimum, 
but we could also repeat the stream multiple time and transform the  other conditionals into filters as well. 

Come to think of it, we could map to a string containing the error message of the exception that we now throw and collect it to a list which then gets merged into a single one with all the validations and returned. I think it would clean up the code that collects all validation failures. 
Right now we only throw these exceptions so we can lather collect and chain them.

Reply:
I pushed an update, we now cumulate even more errors thanks to this change.